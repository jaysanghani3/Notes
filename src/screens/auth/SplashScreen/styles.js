import {StyleSheet} from 'react-native';
import Colors from '../../../utils/Colors';

const styles = StyleSheet.create({
  fundwave: {fontSize: 50, color: Colors.fundwave, fontWeight: '800'},
});

export default styles;
