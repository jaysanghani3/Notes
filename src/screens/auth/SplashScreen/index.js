import {View, Image, Text} from 'react-native';
import React, {useEffect} from 'react';
import auth from '@react-native-firebase/auth';
import globalStyles from '../../../utils/globalStyles';
import styles from './styles';
import Icons from '../../../assets';
import Strings from '../../../translation/Strings';

const SplashScreen = ({navigation}) => {
  useEffect(() => {
    setTimeout(() => {
      const unsubscribe = auth().onAuthStateChanged(user => {
        if (user === null) {
          navigation.replace('LoginScreen');
          return;
        } else if (user.emailVerified) {
          navigation.replace('HomeScreen');
        }
      });
      return unsubscribe;
    }, 2000);
  }, [navigation]);

  return (
    <View style={globalStyles.flexCenter}>
      <Image source={Icons.Logo} />
      <Text style={styles.fundwave}>{Strings.fundwave}</Text>
    </View>
  );
};

export default SplashScreen;
