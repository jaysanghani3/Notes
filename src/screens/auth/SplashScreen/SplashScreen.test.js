import React from 'react';
import renderer from 'react-test-renderer';
import SplashScreen from './index';
import {test, expect} from '@jest/globals';

// jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const tree = renderer.create(<SplashScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
