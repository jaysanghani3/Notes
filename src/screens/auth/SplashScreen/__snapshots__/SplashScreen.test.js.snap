// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/logo.png",
      }
    }
  />
  <Text
    style={
      {
        "color": "#1D5B79",
        "fontSize": 50,
        "fontWeight": "800",
      }
    }
  >
    Fundwave
  </Text>
</View>
`;
